### PR TITLE
If there are overviews, always use them for rendering

### PR DESCRIPTION
**Pull request by rafatower**
I'm currently testing (or trying to) this patch. It is meant to solve part of this issue: https://github.com/CartoDB/cartodb/issues/7120

aside from the correctness/beauty of the patch, opinions are welcomed @rochoa and @javisantana, especially if we want to get this in the upstream repo.

### CODE REVIEW

**Comment by rafatower** [comment]
I tested it and it works as expected.

**Comment by rafatower** [comment]
ping, @rochoa if you don't see any major issue I'll issue a PR to the upstream project. Then we need to bundle it in our tiler.

File: plugins/input/pgraster/pgraster_datasource.cpp [hunk 1]
```diff
@@ -878,8 +878,9 @@ featureset_ptr pgraster_datasource::features_with_context(query const& q,process
         std::string col = geometryColumn_;
 
         if ( use_overviews_ ) {
-          std::string sch = schema_;
-          std::string tab = mapnik::sql_utils::unquote_double(raster_table_);
+          std::string sch = overviews_[0].schema;
```
Comment (rochoa) [inline plugins/input/pgraster/pgraster_datasource.cpp:881]: `use_overviews_` might be true but you might not have overviews.

`use_overviews_` is a configuration parameter to determine if overview tables must be used or not. We at least should check `overviews_[0]` is defined.

**Comment by rafatower** [comment]
I'm closing this PR and working on another branch to polish the changes and issue a proper PR in the upstream repo (if you see my latest commit added files that I did not really intend to).

**Comment by jgoizueta** [comment]
I had this (crazy?) idea:

If, after importing a raster table `raster_columns` is altered for the overview of factor=1 to change `scale_x` and `scale_y` to 0, then no changes would be needed in Mapnik to always use the factor=1 overview as default.

**Comment by rafatower** [comment]
@jgoizueta reasons not to, without diving any further: https://github.com/mapnik/mapnik/pull/2589

but thanks anyway

**Pull request closed without merging**